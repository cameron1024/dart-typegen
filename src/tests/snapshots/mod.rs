// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
