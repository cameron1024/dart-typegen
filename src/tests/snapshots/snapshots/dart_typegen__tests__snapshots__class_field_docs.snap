---
source: src/tests/snapshots/mod.rs
expression: dart
---
import "package:equatable/equatable.dart";

final class Foo with EquatableMixin {
  /// single line doc
  final String x;

  /// A multi-line doc
  ///
  /// Some more stuff
  final String y;

  const Foo({required this.x, required this.y});

  @override
  List<Object?> get props => [x, y];

  FooBuilder toBuilder() => FooBuilder(x: x, y: y);

  Map<String, dynamic> toJson() => {"x": x, "y": y};
  factory Foo.fromJson(Map<String, dynamic> json) =>
      Foo(x: json["x"] as String, y: json["y"] as String);

  @override
  String toString() =>
      "Foo("
      "x: $x,"
      "y: $y"
      ")";
}

/// Builder class for [Foo]
final class FooBuilder {
  String x;
  String y;

  FooBuilder({required this.x, required this.y});

  Foo build() => Foo(x: x, y: y);
}
